### PR TITLE
Add webhook logging to new webhooks handler

### PR DESCRIPTION
**Pull request by mollylogue**
- add webhook logging middleware to new webhooks handler to log webhooks and ensure webhook ID is set on log
- expand webhook_logs db methods to include new webhook_id column
- Note: generally followed the existing patterns for the existing integration with external services. Eventually we'll deprecate that in favor of just connecting the logs to the webhook id.
- part of https://github.com/sourcegraph/sourcegraph/issues/43105

## Test plan
Existing/expanded unit tests pass

### CODE REVIEW

File: cmd/frontend/webhooks/middleware.go [hunk 1]
```diff
@@ -20,7 +20,17 @@ func SetExternalServiceID(ctx context.Context, id int64) {
...
+		setter(id, 0)
+	}
+}
+
+// SetWebhookID attaches a specific webhook ID to the current
+// webhook request for logging purposes.
+func SetWebhookID(ctx context.Context, id int32) {
+	// There's no else case here because it is expected that there's no setter
+	// if logging is disabled.
+	if setter, ok := ctx.Value(setterContextKey).(contextFunc); ok {
```
Comment (ryanslade) [inline cmd/frontend/webhooks/middleware.go:32]: I think we should separate it out into two setter functions, one for each id. 

Then we don't need to do the checks if one value is zero and it's more explicit what each setter is doing.

File: cmd/frontend/webhooks/webhooks.go [hunk 2]
```diff
@@ -24,7 +24,10 @@ type WebhookHandlers struct {
 // come from.
 func NewHandler(logger log.Logger, db database.DB, gh *GitHubWebhook) http.Handler {
 	logger = logger.Scoped("webhookHandler", "handler used to route webhooks")
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	webhookMiddleware := NewLogMiddleware(
```
Comment (ryanslade) [inline cmd/frontend/webhooks/webhooks.go:27]: Probably better to do this here:

https://github.com/sourcegraph/sourcegraph/blob/main/cmd/frontend/internal/httpapi/httpapi.go#L104

Comment (ryanslade) [reply cmd/frontend/webhooks/webhooks.go:27]: Since at that point we already have the middleware and it's where we wrap the other handlers.

**Review by ryanslade — APPROVED** [review]
Nice, just a few small comments

File: cmd/frontend/webhooks/middleware.go (hunk 1 above)
Comment (mollylogue) [reply cmd/frontend/webhooks/middleware.go:32]: Sounds good 👍

Comment (mollylogue) [reply cmd/frontend/webhooks/middleware.go]: So, I've done that in [this commit](https://github.com/sourcegraph/sourcegraph/pull/43446/commits/ed4dbf1d82347cde47fffb4f22456babe00fd226) lmk if you think this version is better or not.

File: cmd/frontend/webhooks/webhooks.go (hunk 2 above)
Comment (mollylogue) [reply cmd/frontend/webhooks/webhooks.go:27]: 👍

File: cmd/frontend/webhooks/middleware.go [hunk 3]
```diff
@@ -63,17 +73,22 @@ func (mw *LogMiddleware) Logger(next http.Handler) http.Handler {
...
 		// we need access to it to be able to store the webhook log with the
-		// appropriate external service ID. To handle this, we'll put a setter
+		// appropriate external service/webhook ID. To handle this, we'll put a setter
 		// closure into the context that can then be used by
 		// SetExternalServiceID to receive the external service ID from the
 		// handler.
 		var externalServiceID *int64
-		var setter contextFunc = func(id int64) {
-			externalServiceID = &id
+		var extSvcIDSetter contextFuncInt64 = func(extSvcID int64) {
```
Comment (ryanslade) [inline cmd/frontend/webhooks/middleware.go:83]: nit

```suggestion
		var extSvcIDSetter = func(extSvcID int64) {
```

File: cmd/frontend/webhooks/middleware.go [hunk 4]
```diff
@@ -63,17 +73,22 @@ func (mw *LogMiddleware) Logger(next http.Handler) http.Handler {
...
+			externalServiceID = &extSvcID
+		}
+		ctx := context.WithValue(r.Context(), extSvcIDSetterContextKey, extSvcIDSetter)
+		var webhookID *int32
+		var webhookIDSetter contextFuncInt32 = func(whID int32) {
```
Comment (ryanslade) [inline cmd/frontend/webhooks/middleware.go:88]: nit

```suggestion
		var webhookIDSetter = func(whID int32) {
```

**Review by ryanslade — APPROVED** [review]
LGTM

File: cmd/frontend/webhooks/middleware.go (hunk 3 above)
Comment (mollylogue) [reply cmd/frontend/webhooks/middleware.go:83]: So, when I remove the type, all the tests panic 🤔 I'm trying to figure out why. I'm guessing this is why the original code specified the type when setting this variable. This is likely some Go-ism that I don't fully understand

Comment (mollylogue) [reply cmd/frontend/webhooks/middleware.go:83]: (and by all the tests I mean the tests specific to this code path 😅 )

**Pull request merged**